tests/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM python:3.9-slim as builder
+FROM python:3.12-slim as builder
```

Comment:
@oren-zohar The command `just build-pytest-docker` works because the `pyproject.toml` file still exists in the `tests` folder. This `Dockerfile` copies `pyproject.toml` to create the `requirements.txt` file. If `pyproject.toml` is removed, K8s CI tests will not work.